integration-tests/java/src/integration-test/java/com/stellarstation/api/test/satellite/SatelliteStreamerTest.java

```diff
@@ -124,19 +115,83 @@ public void onCompleted() {}
             .setEnableEvents(true)
             .build());
 
-    ListenableFuture<Boolean> timeoutFuture =
-        Futures.withTimeout(
-            commandTestFuture, 30, TimeUnit.SECONDS, Executors.newSingleThreadScheduledExecutor());
+    TimeUnit.SECONDS.sleep(20);
```

Comment:
I'm wondering if we can get rid of this delay with a slightly different approach.

Inside onNext, I assume we can detect when the test has collected all data that we need. Maybe we can call requestObserverFuture.onCompleted directly from onNext?

If we do that, then the main thread just needs to wait for the timeoutFuture without any calls to sleep.